tests/cmake.py

```diff
@@ -102,9 +103,23 @@ def cmake(cwd, targets, options=None):
                 "ANDROID_NATIVE_API_LEVEL": os.environ["ANDROID_API"],
             }
         )
+
+    source_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
     cmake = ["cmake"]
     if "scan-build" in os.environ.get("RUN_ANALYZER", ""):
-        cmake = ["scan-build", "cmake"]
+        cc = os.environ.get("CC")
+        cxx = os.environ.get("CXX")
+        cmake = [
+            "scan-build",
+            *(["--use-cc", cc] if cc else []),
+            *(["--use-c++", cxx] if cxx else []),
+            "--status-bugs",
+            "--exclude",
+            os.path.join(source_dir, "external"),
+            "cmake",
+        ]
+
     configcmd = [*cmake]
```

Comment:
`cmake` here is always a list, why the weird `[*cmake]` syntax instead of directly `configcmd = cmake`?

Reply:
I use it another time below, I wasn’t sure if `configcmd = cmake` just creates a reference (that I mutate) or actually creates a copy. I basically force it to be a copy, so I can mutate without having problems later.

Reply:
Oh, yes this would be just a reference.  To force a copy `configcmd = cmake.copy()` is much clearer and more pythonic imho.